src/models/common/metadata.ts

```diff
@@ -0,0 +1,38 @@
+import { Amount } from ".";
+
+interface CreatedNode {
+    CreatedNode: {
+        LedgerEntryType: string
+        LedgerIndex: string
+        NewFields: {[field: string]: any}
+    }
+}
+
+interface ModifiedNode {
+    ModifiedNode: {
+        LedgerEntryType: string
+        LedgerIndex: string
+        FinalFields: {[field: string]: any}
+        PreviousFields: {[field: string]: any}
+        PreviousTxnID?: string
+        PreviouTxnLgrSeq?: number
+    }
+}
+
+interface DeletedNode {
+    DeletedNode: {
+        LedgerEntryType: string
+        LedgerIndex: string
+        FinalFields: {[field: string]: any}
+    }
+}
+
+type Node = CreatedNode | ModifiedNode | DeletedNode
+
+export interface Metadata {
```

Comment:
Is the name `Metadata` potentially ambiguous? Is there any other place where metadata is relevant in rippled?

Reply:
I dont think so, I mostly didn't want to have types `TransactionMetaData` and `TransactionAndMetaData`

Reply:
Default export